libqtile/widget/battery.py

```diff
@@ -344,57 +362,115 @@ def set_battery_charge_thresholds(self, start, end):
+                    if stat == "Full":
+                        state = BatteryState.FULL
+                    elif stat == "Charging":
+                        state = BatteryState.CHARGING
+                    elif stat == "Dischaging":
+                        state = BatteryState.DISCHARGING
+                    elif stat == "Not charging":
+                        state = BatteryState.NOT_CHARGING
+                    else:
+                        pass  # other status values keep the value of state
+                elif state == BatteryState.NOT_CHARGING:
+                    if stat == "Full":
+                        state = BatteryState.FULL
+                    if stat == "Charging":
+                        state = BatteryState.CHARGING
+                    if stat == "Discharging":
+                        state = BatteryState.DISCHARGING
+                    if stat == "Empty":
+                        state = BatteryState.EMPTY
+                    else:
+                        pass  # other status values keep the value of state
+                else:
+                    if stat == "Full":
+                        state = BatteryState.FULL
+                    if stat == "Charging":
+                        state = BatteryState.CHARGING
+                    if stat == "Discharging":
+                        state = BatteryState.DISCHARGING
+                    if stat == "Empty":
+                        state = BatteryState.EMPTY
+                    else:
+                        pass  # other status values keep the value of state
+
+                now_str, now_unit = self._get_param("energy_now_file", bat)
+                full_str, full_unit = self._get_param("energy_full_file", bat)
+                power_str, power_unit = self._get_param("power_now_file", bat)
+                # the units of energy is uWh or uAh, multiply to get to uWs or uAs
+                now = 3600 * float(now_str)
+                full = 3600 * float(full_str)
+                power = float(power_str)
+
+                if now_unit != full_unit:
+                    raise RuntimeError("Current and full energy units do not match")
+                if full == 0:
+                    percent += 0.0
+                else:
+                    percent += now / full
+
+                if power == 0:
+                    time += 0
+                elif state == BatteryState.DISCHARGING:
+                    time += int(now / power)
+                else:
+                    time += int((full - now) / power)
+
+                if power_unit == "uA":
+                    voltage = float(self._get_param("voltage_now_file", bat)[0])
+                    power = voltage * power / 1e12
+                elif power_unit == "uW":
+                    power = power / 1e6
+
+                power_acc += power
 
         return BatteryStatus(
             state=state,
-            percent=percent,
-            power=power,
+            percent=percent / len(bats),
```

Comment:
This assumes batteries have equal capacity. The percentage should be calculated as the sum of `now` over the sum of `full`.

Reply:
Hi, Sorry for the very late reply
I fixed the formula in linux and freebsd. The value of **now** in freebsd can not be got so easy as in linux, the only way to do it was usingpercentage and the full value of each battery. The results are consistent using other ways to check the total percentage, gnome battery widget in linux or apm command in freebsd